tony-core/src/main/java/com/linkedin/tony/events/EventHandler.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright 2018 LinkedIn Corporation. All rights reserved. Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.tony.events;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.linkedin.tony.Constants;
+import com.linkedin.tony.TonyJobMetadata;
+import com.linkedin.tony.util.HistoryFileUtils;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.concurrent.BlockingQueue;
+import org.apache.avro.file.DataFileWriter;
+import org.apache.avro.io.DatumWriter;
+import org.apache.avro.specific.SpecificDatumWriter;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+
+public class EventHandler implements Runnable {
+  private static final Log LOG = LogFactory.getLog(EventHandler.class);
+  private boolean isStopped = false;
+  private BlockingQueue<Event> eventQueue;
+  private Path historyFile = null;
+  private Path tmpFile = new Path(Constants.TMP_AVRO);
+  private DatumWriter<Event> eventWriter = new SpecificDatumWriter<>();
+  private DataFileWriter<Event> dataFileWriter = new DataFileWriter<>(eventWriter);
+  private OutputStream out;
+  private FileSystem myFs;
+
+  public EventHandler(FileSystem fs, BlockingQueue<Event> q) {
+    eventQueue = q;
+    myFs = fs;
+    try {
+      out = myFs.create(tmpFile);
+      dataFileWriter.create(Event.SCHEMA$, out);
+    } catch (IOException e) {
+      LOG.error("Failed to set up writer", e);
+    }
+  }
+
+  public void emitEvent(Event event) {
+    try {
+      eventQueue.put(event);
+    } catch (InterruptedException e) {
+      LOG.error("Failed to add event " + event + " to event queue", e);
+    }
+  }
+
+  @VisibleForTesting
+  public void writeEvent(BlockingQueue<Event> queue, DataFileWriter<Event> writer) {
+    Event event = null;
+    try {
+      event = queue.poll();
```

Comment:
This should be switched to .take() and we should use interrupt to stop the event handler.

Currently, we are doing a lot of spinning:
```
while (!isStopped) {
  writeEvent(...);
}
``` 

is going to use a lot of CPU looping over and over again.

Reply:
Filed https://github.com/linkedin/TonY/issues/128 for this.